specreduce/wavelength_calibration.py

```diff
@@ -184,67 +183,59 @@ def input_spectrum(self, new_spectrum):
         self._input_spectrum = new_spectrum
 
     @property
-    def model(self):
-        return self._model
+    def input_model(self):
+        return self._input_model
 
-    @model.setter
-    def model(self, new_model):
+    @input_model.setter
+    def input_model(self, input_model):
         self._clear_cache()
-        self._model = new_model
+        self._input_model = input_model
 
     @cached_property
-    def fit_residuals(self):
-        # calculate fit residuals between matched line list pixel centers and
-        # wavelengths and the evaluated fit model. this is only accessible
-        # after the ``wcs`` attribute is accessed, where the fit is actually
-        # performed. ``apply_to_spectrum`` also calls .wcs and therefore
-        # performs the fit, so this is also acessible if that is called.
-
-        x = self._matched_line_list["pixel_center"]
-        y = self._matched_line_list["wavelength"]
-
-        if self._fit_resids is None:
-            raise ValueError('Fit residuals are only available after the new'
-                             ' WCS is fit - this can be done by accessing the'
-                             ' ``.wcs`` attribute, or by calling the'
-                             ' ``.apply_to_spectrum`` method.')
-
-        # Get the fit residuals by evaulating model
-        self._fit_resids = y - self._model(x)
-
-        return self._fit_resids
-
-    @cached_property
-    def wcs(self):
+    def fitted_model(self):
         # computes and returns WCS after fitting self.model to self.refined_pixels
         x = self._matched_line_list["pixel_center"]
         y = self._matched_line_list["wavelength"]
 
         if self.fitter is None:
             # Flexible defaulting if self.fitter is None
-            if self.model.linear:
+            if self.input_model.linear:
                 fitter = LinearLSQFitter(calc_uncertainties=True)
             else:
                 fitter = LMLSQFitter(calc_uncertainties=True)
         else:
             fitter = self.fitter
 
         # Fit the model
-        self._model = fitter(self._model, x, y)
+        return fitter(self.input_model, x, y)
+
+    @cached_property
+    def residuals(self):
+        """
+        calculate fit residuals between matched line list pixel centers and
```

Comment:
this docstring is no longer true since accessing this will cause the model to be fit. i think you could just leave this at "calculate fit residuals between matched line list pixel centers and wavelengths and the evaluated fit model."